paderbox/array/sparse.py

```diff
@@ -397,16 +397,51 @@ def from_array_and_onset(cls, array, onset, shape):
         array([1., 1., 0., 0., 0., 0., 0., 0., 0., 0.])
         >>> SparseArray.from_array_and_onset(np.ones(5), 7, 10)
         SparseArray(_SparseSegment(onset=7, array=array([1., 1., 1.])), shape=(10,))
+        >>> SparseArray.from_array_and_onset(np.ones(5), 7).as_contiguous()
+        array([0., 0., 0., 0., 0., 0., 0., 1., 1., 1., 1., 1.])
         """
+        if shape is None:
+            shape = list(array.shape)
+            shape[-1] += onset
+            if shape[-1] < 0:
+                shape[-1] = 0
         out = SparseArray(shape=shape)
-        if onset < 0:
-            array = array[..., -onset:]
-            onset = 0
-        if array.shape[-1] + onset > out.shape[-1]:
-            array = array[..., :out.shape[-1] - onset]
-        out._add_segment(_SparseSegment(onset, array))
+        out._add_segment(_SparseSegment.from_array(array, onset, out.shape[-1]))
+        return out
+
+    @staticmethod
+    def from_arrays_and_onsets(arrays, onsets, shape=None, pad_value=0.):
+        """
+        >>> SparseArray.from_arrays_and_onsets([np.ones(2), np.ones(1)], [0, 3]).as_contiguous()
+        array([1., 1., 0., 1.])
+        >>> SparseArray.from_arrays_and_onsets([np.ones(2), np.ones(1)], [0, 3], (10,)).as_contiguous()
+        array([1., 1., 0., 1., 0., 0., 0., 0., 0., 0.])
+        """
+        if shape is None:
+            for array, onset in zip(arrays, onsets):
+                if shape is None or array.shape[-1] + onset > shape[-1]:
+                    shape = list(array.shape)
+                    shape[-1] += onset
```

Comment:
Suggestion: Add helper function, that predicts the shape.

Why? `from_array_and_onset` has
```
            if shape[-1] < 0:
                shape[-1] = 0
```
while it is missing in `from_arrays_and_onsets`.